core/src/com/mygdx/game/Player.java

```diff
@@ -149,13 +149,20 @@ public void tick()
         }
         if (Gdx.input.isKeyPressed(Input.Keys.E)) {// RECOGER
             for (int i = 0; i < level.getItems().size(); i++ ) {
-                if (hitbox.overlaps(level.getItems().get(i).hitbox))
-                {
-                    double now = System.nanoTime();
-                    if (now - lastHit > 1000000000) {
-                        System.out.println(2);
-                        lastHit = now;
-                        level.getItems().remove(i);
+                for (int j = 0; j < level.getItems().get(i).size(); j++) {
+                    if (hitbox.overlaps(level.getItems().get(i).get(j).hitbox)
+                            && level.getItems().get(i).get(j).getPickable() == 1) {
+                        double now = System.nanoTime();
+                        if (now - lastHit > 1000000000) {
+                            //System.out.println(2);
+                            lastHit = now;
+                            level.getItems().get(i).get(j).setPickable(0);
+                            for (int k = 0; k < level.getGroup().size(); k++) {
+                                if (level.getGroup().get(k).getIndexList() == i) {
+                                    level.getGroup().get(k).setCounter(level.getGroup().get(k).getCounter() - 1);
+                                }
+                            }
+                        }
```

Comment:
Por legibilidad yo usaría un _enhanced for-loop_, pero es nomas una recomendación.
[Ejemplo](https://www.programcreek.com/2012/10/loop-a-iterable-in-java/)

Reply:
Bueno voy viendo que el _i_ si se necesita, pero el segundo for creo que si se puede simplificar.
Creo.